ribs/visualize.py

```diff
@@ -114,32 +152,19 @@ def grid_archive_heatmap(archive,
         pcm_kwargs (dict): Additional kwargs to pass to
             :func:`~matplotlib.pyplot.pcolormesh`.
         cbar_kwargs (dict): Additional kwargs to pass to :func:`~matplotlib.figure.Figure.colorbar`
+
     Raises:
-        ValueError: The archive is not 2D.
+        ValueError: The archive's dimension is not supported (not 1D or 2D).
     """
-    if square is not None:
-        raise ValueError(
-            "The argument 'square' is deprecated and will not be "
-            "supported in future versions. Use 'aspect' to set the "
-            "heatmap's aspect ratio instead")
-    if archive.behavior_dim not in [1, 2]:
-        raise ValueError("Heatmaps are only supported for 1D and 2D archives")
-    if not (cbar == "auto" or isinstance(cbar, axes.Axes) or cbar is None):
-        raise ValueError(
-            f"Invalid arg cbar={cbar}; must be 'auto', None, or matplotlib.axes.Axes"
-        )
-
     if aspect is None:
         # Handles default aspects for different dims.
         if archive.behavior_dim == 1:
             aspect = 0.5
         else:
             aspect = "auto"
 
-    if aspect is not None and not (isinstance(aspect, float) or
-                                   aspect in ["equal", "auto"]):
-        raise ValueError(
-            f"Invalid arg aspect='{aspect}'; must be 'auto', 'equal', or float")
+    _validate_heatmap_visual_args(aspect, cbar, square, archive.behavior_dim,
```

Comment:
I think this check should go at the very beginning of the function. Same for the cvt archive heatmap.